CPUserService/src/main/java/com/carpooling/ums/controllers/PermissionController.java

```diff
@@ -0,0 +1,79 @@
+
+    private @Autowired PermissionService permissionService;
+    private static final Logger logger = LoggerFactory.getLogger(PermissionController.class);
+
+    @GetMapping
+    public ResponseEntity<ApiResponse<List<PermissionDTO>>> getAllPermissions() {
+        try {
+            List<Permission> permissions = permissionService.getAllPermissions();
+            List<PermissionDTO> permissionDTOs = DtoConverter.convertToDtoList(permissions, PermissionDTO.class);
+            return ResponseEntity.ok(new ApiResponse<>(true, "Permissions retrieved successfully", permissionDTOs));
+        } catch (Exception e) {
+            logger.error("Error retrieving permissions", e);
+            return ResponseEntity.status(500).body(new ApiResponse<>(false, "Internal Server Error", null));
+        }
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<ApiResponse<PermissionDTO>> getPermissionById(@PathVariable Long id) {
+        try {
+            Permission permission = permissionService.getPermissionById(id);
+            if (permission != null) {
+                PermissionDTO permissionDTO = DtoConverter.convertToDto(permission, PermissionDTO.class);
+                return ResponseEntity.ok(new ApiResponse<>(true, "Permission retrieved successfully", permissionDTO));
+            } else {
+                return ResponseEntity.status(404).body(new ApiResponse<>(false, "Permission not found", null));
+            }
+        } catch (Exception e) {
+            logger.error("Error retrieving permission with id: {}", id, e);
+            return ResponseEntity.status(500).body(new ApiResponse<>(false, "Internal Server Error", null));
+        }
+    }
+
+    @PostMapping
+    public ResponseEntity<ApiResponse<PermissionDTO>> createPermission(@RequestBody PermissionDTO permissionDTO) {
+        try {
+            Permission permission = DtoConverter.convertToEntity(permissionDTO, Permission.class);
+            Permission createdPermission = permissionService.createPermission(permission);
+            PermissionDTO createdPermissionDTO = DtoConverter.convertToDto(createdPermission, PermissionDTO.class);
+            return ResponseEntity.status(201).body(new ApiResponse<>(true, "Permission created successfully", createdPermissionDTO));
+        } catch (Exception e) {
+            logger.error("Error creating permission", e);
+            return ResponseEntity.status(500).body(new ApiResponse<>(false, "Internal Server Error", null));
+        }
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<ApiResponse<Void>> deletePermission(@PathVariable Long id) {
+        try {
+            boolean deleted = permissionService.deletePermission(id);
+            if (deleted) {
+                return ResponseEntity.ok(new ApiResponse<>(true, "Permission deleted successfully", null));
+            } else {
+                return ResponseEntity.status(404).body(new ApiResponse<>(false, "Permission not found", null));
```

Comment:
Error message should be thrown from service layer and catched here and to sent in response